examples/contrib/gp/sv-dkl.py

```diff
@@ -0,0 +1,137 @@
+import pyro.optim as optim
+from examples import util
+
+
+class CNN(nn.Module):
+    def __init__(self):
+        super(CNN, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2(x), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = self.fc2(x)
+        return x
+
+
+def train(args, train_loader, gpmodel, svi, epoch):
+    for batch_idx, (data, target) in enumerate(train_loader):
+        if args.cuda:
+            data, target = data.cuda(), target.cuda()
+        gpmodel.set_data(data, target)
+        loss = svi.step()
+        if batch_idx % args.log_interval == 0:
+            print('Train Epoch: {:2d} [{:5d}/{} ({:2.0f}%)]\tLoss: {:.6f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                100. * batch_idx / len(train_loader), loss))
+
+
+def test(args, test_loader, gpmodel):
+    correct = 0
+    for data, target in test_loader:
+        if args.cuda:
+            data, target = data.cuda(), target.cuda()
+        f_loc, f_var = gpmodel(data)
+        pred = gpmodel.likelihood(f_loc, f_var)
+        correct += pred.eq(target).long().cpu().sum()
+
+    print('\nTest set: Accuracy: {}/{} ({:.0f}%)\n'.format(
+        correct, len(test_loader.dataset),
+        100. * correct / len(test_loader.dataset)))
+
+
+def main(args):
+    train_loader = util.get_data_loader(dataset_name='MNIST',
+                                        batch_size=args.batch_size,
+                                        dataset_transforms=[transforms.Normalize((0.1307,), (0.3081,))],
+                                        is_training_set=True,
+                                        shuffle=True)
+    test_loader = util.get_data_loader(dataset_name='MNIST',
+                                       batch_size=args.batch_size,
+                                       dataset_transforms=[transforms.Normalize((0.1307,), (0.3081,))],
+                                       is_training_set=False,
+                                       shuffle=True)
+
+    cnn = CNN().cuda() if args.cuda else CNN()
+
+    # helper to inject CNN's parameters to gpmodel
+    def cnn_fn(x):
+        return pyro.module("CNN", cnn)(x)
+    # create deep kernel by warping RBF with cnn
+    kernel = gp.kernels.RBF(input_dim=10, lengthscale=torch.ones(10)).warp(iwarping_fn=cnn_fn)
+
+    # init inducing points (taken randomly from dataset)
+    Xu = next(iter(train_loader))[0][:args.num_inducing]
+    likelihood = gp.likelihoods.MultiClass(num_classes=10)
+    gpmodel = gp.models.SparseVariationalGP(X=Xu, y=None, kernel=kernel, Xu=Xu,
+                                            likelihood=likelihood, latent_shape=torch.Size([10]),
```

Comment:
Add comment explaining why you're using this likelihood and this model, how `latent_shape` is related to `likelihood`, and what whitening is and why you're using it here